torchcrf/__init__.py

```diff
@@ -102,23 +102,30 @@ def forward(
         self._validate(emissions, mask)
 
         seq_length, batch_size = tags.shape
-        # Start transition score and first emission for both
+
+        # Start transition score and first emission
         # shape: (batch_size,)
         numerator = self.start_transitions[tags[0]]
         numerator += emissions[0, torch.arange(batch_size), tags[0]]
-        denominator = self.start_transitions + emissions[0]
+        # Transition score and emission to next tag, only added if next timestep is valid
+        # shape: (batch_size,)
+        numerator += ((
+            emissions[torch.arange(1, seq_length).view(-1, 1).repeat(1, batch_size),
+                      torch.arange(batch_size).repeat(seq_length - 1, 1), tags[1:]] +
+            self.transitions[tags[:-1], tags[1:]]) * mask[1:].float()).sum(dim=0)
```

Comment:
This looks rather complicated. I prefer the old one for readability, could you please revert it back?